base/applications/cleanmgr/util.c

```diff
@@ -0,0 +1,799 @@
+		switch (ListView_GetCheckState(hList, index))
+		{
+		case TRUE:
+		{
+			bv.tempClean = TRUE;
+			size += tempSize;
+			setTotalSize(size, IDC_STATIC_SIZE, hwnd);
+		}
+		return size;
+
+		case FALSE:
+		{
+			bv.tempClean = FALSE;
+			size -= tempSize;
+			if (0 >= size)
+			{
+				SetDlgItemTextW(hwnd, IDC_STATIC_SIZE, L"0.00 Bytes");
+				return 0;
+			}
+			setTotalSize(size, IDC_STATIC_SIZE, hwnd);
+		}
+		return size;
+		}
+
+	case RECYCLE_BIN:
+		switch (ListView_GetCheckState(hList, index))
+		{
+		case TRUE:
+		{
+			bv.recycleClean = TRUE;
+			size += recycleSize;
+			setTotalSize(size, IDC_STATIC_SIZE, hwnd);
+		}
+		return size;
+
+		case FALSE:
+		{
+			bv.recycleClean = FALSE;
+			size -= recycleSize;
+			if (0 >= size)
+			{
+				SetDlgItemTextW(hwnd, IDC_STATIC_SIZE, L"0.00 Bytes");
+				return 0;
+			}
+			setTotalSize(size, IDC_STATIC_SIZE, hwnd);
+		}
+		return size;
+		}
+
+	case DOWNLOADED_FILES:
+		switch (ListView_GetCheckState(hList, index))
+		{
+		case TRUE:
+		{
+			bv.downloadClean = TRUE;
+			size += downloadSize;
+			setTotalSize(size, IDC_STATIC_SIZE, hwnd);
+		}
+		return size;
+
+		case FALSE:
+		{
+			bv.downloadClean = FALSE;
+			size -= downloadSize;
+			if (0 >= size)
+			{
+				SetDlgItemTextW(hwnd, IDC_STATIC_SIZE, L"0.00 Bytes");
+				return 0;
+			}
+			setTotalSize(size, IDC_STATIC_SIZE, hwnd);
+		}
+		return size;
+		}
+
+	case RAPPS_FILES:
+		switch (ListView_GetCheckState(hList, index))
+		{
+		case TRUE:
+		{
+			bv.rappsClean = TRUE;
+			size += rappsSize;
+			setTotalSize(size, IDC_STATIC_SIZE, hwnd);
+		}
+		return size;
+
+		case FALSE:
+		{
+			bv.rappsClean = FALSE;
+			size -= rappsSize;
+			if (0 >= size)
+			{
+				SetDlgItemTextW(hwnd, IDC_STATIC_SIZE, L"0.00 Bytes");
+				return 0;
+			}
+			setTotalSize(size, IDC_STATIC_SIZE, hwnd);
+		}
+		return size;
+		}
+
+	default:
+		return 0;
+	}
+}
+
+LRESULT APIENTRY ThemeHandler(HWND hDlg, NMCUSTOMDRAW* pNmDraw)
+{
+	HTHEME hTheme;
+	HWND hDlgButtonCtrl;
+	LRESULT Ret;
+	INT iState = PBS_NORMAL;
+
+	hDlgButtonCtrl = pNmDraw->hdr.hwndFrom;
+	hTheme = GetWindowTheme(hDlgButtonCtrl);
+
+	if (hTheme)
+	{
+		switch (pNmDraw->dwDrawStage)
+		{
+		case CDDS_PREPAINT:
+		{
+			if (pNmDraw->uItemState & CDIS_DEFAULT)
+			{
+				iState = PBS_DEFAULTED;
+			}
+			else if (pNmDraw->uItemState & CDIS_SELECTED)
+			{
+				iState = PBS_PRESSED;
+			}
+			else if (pNmDraw->uItemState & CDIS_HOT)
+			{
+				iState = PBS_HOT;
+			}
+
+			if (IsThemeBackgroundPartiallyTransparent(hTheme, BP_PUSHBUTTON, iState))
+			{
+				DrawThemeParentBackground(hDlgButtonCtrl, pNmDraw->hdc, &pNmDraw->rc);
+			}
+
+			DrawThemeBackground(hTheme, pNmDraw->hdc, BP_PUSHBUTTON, iState, &pNmDraw->rc, NULL);
+
+			Ret = CDRF_SKIPDEFAULT;
+			break;
+		}
+
+		case CDDS_PREERASE:
+		{
+			Ret = CDRF_DODEFAULT;
+			break;
+		}
+
+		default:
+			Ret = CDRF_SKIPDEFAULT;
+			break;
+		}
+	}
+	else
+	{
+		Ret = CDRF_DODEFAULT;
+	}
+
+	return Ret;
+}
+
+BOOL EnableDialogTheme(HWND hwnd)
```

Comment:
Don't change the theme in generic tool.

Reply:
@katahiromz This is actually for child windows so its controls don't look inconsistent (i.e. some controls having completely different color compared to child window). Sorry for late reply.

Reply:
is this needed on windows?

Reply:
@learn-more Yes, also for ReactOS too. By the way, if you are wondering where did I get to know about the working of tabbed dialog box, I got it from `msconfig`.